scipy/linalg/tests/test_decomp_cholesky.py

```diff
@@ -1,69 +1,104 @@
+        c = c.T
+        a = c @ c.T
+        xp_assert_close(cholesky(a, lower=True, check_finite=False), c)
+
+    # https://github.com/numpy/numpy/issues/24451
+    @skip_if_array_api_backend('numpy.array_api')
+    # https://github.com/data-apis/array-api-compat/issues/54
+    @skip_if_array_api_backend('cupy')
+    @array_api_compatible
+    def test_simple_complex(self, xp):
+        m = xp.asarray([[3+1j, 3+4j, 5], [0, 2+2j, 2+7j], [0, 0, 7+4j]])
+        a = xp.conj(m).T @ m
         c = cholesky(a)
-        a1 = dot(transpose(conjugate(c)), c)
-        assert_array_almost_equal(a, a1)
-        c = transpose(c)
-        a = dot(c, transpose(conjugate(c)))
-        assert_array_almost_equal(cholesky(a, lower=1), c)
-
-    def test_random(self):
+        a1 = xp.conj(c).T @ c
+        xp_assert_close(a, a1)
+        c = c.T
+        a = c @ xp.conj(c).T
+        xp_assert_close(cholesky(a, lower=True), c)
+
+    @array_api_compatible
+    def test_random(self, xp):
         n = 20
         for k in range(2):
             m = random([n, n])
             for i in range(n):
                 m[i, i] = 20*(.1+m[i, i])
-            a = dot(transpose(m), m)
+            m = xp.asarray(m)
+            a = m.T @ m
             c = cholesky(a)
-            a1 = dot(transpose(c), c)
-            assert_array_almost_equal(a, a1)
-            c = transpose(c)
-            a = dot(c, transpose(c))
-            assert_array_almost_equal(cholesky(a, lower=1), c)
-
-    def test_random_complex(self):
+            a1 = c.T @ c
+            xp_assert_close(a, a1)
+            c = c.T
+            a = c @ c.T
+            xp_assert_close(cholesky(a, lower=True), c)
+
+    # https://github.com/numpy/numpy/issues/24451
+    @skip_if_array_api_backend('numpy.array_api')
+    # https://github.com/data-apis/array-api-compat/issues/54
+    @skip_if_array_api_backend('cupy')
+    @array_api_compatible
+    def test_random_complex(self, xp):
         n = 20
         for k in range(2):
             m = random([n, n])+1j*random([n, n])
             for i in range(n):
                 m[i, i] = 20*(.1+abs(m[i, i]))
-            a = dot(transpose(conjugate(m)), m)
+            m = xp.asarray(m)
+            a = xp.conj(m).T @ m
             c = cholesky(a)
-            a1 = dot(transpose(conjugate(c)), c)
-            assert_array_almost_equal(a, a1)
-            c = transpose(c)
-            a = dot(c, transpose(conjugate(c)))
-            assert_array_almost_equal(cholesky(a, lower=1), c)
+            a1 = xp.conj(c).T @ c
+            xp_assert_close(a, a1)
+            c = c.T
+            a = c @ xp.conj(c).T
+            xp_assert_close(cholesky(a, lower=True), c)
+
+    @array_api_compatible
+    @pytest.mark.parametrize("dtype", ["float32", "float64"])
+    def test_dtypes_standard(self, dtype, xp):
+        a = xp.asarray([[8, 2, 3], [2, 9, 3], [3, 3, 6]], dtype=getattr(xp, dtype))
+        c = cholesky(a)
+        rtol = 1e-7 if dtype == "float64" else 1e-6
```

Comment:
this variation of rtol by dtype could be used elsewhere?

Reply:
You may just want to use floating point properties themselves. For instance,

```
>>> np.finfo(np.float32).eps
1.1920929e-07
>>> np.finfo(np.float64).eps
2.220446049250313e-16
```

I would normally use the above to get a tolerance that varies for different floating point types. But in your case it only differs by a single order of magnitude?